pandas/core/nanops.py

```diff
@@ -425,65 +425,34 @@ def nansem(values, axis=None, skipna=True, ddof=1):
     return np.sqrt(var) / np.sqrt(count)
 
 
-@bottleneck_switch()
-def nanmin(values, axis=None, skipna=True):
-    values, mask, dtype, dtype_max = _get_values(values, skipna,
-                                                 fill_value_typ='+inf')
-
-    # numpy 1.6.1 workaround in Python 3.x
-    if is_object_dtype(values) and compat.PY3:
-        if values.ndim > 1:
-            apply_ax = axis if axis is not None else 0
-            result = np.apply_along_axis(builtins.min, apply_ax, values)
-        else:
-            try:
-                result = builtins.min(values)
-            except:
-                result = np.nan
-    else:
+def _nanminmax(meth, fill_value_typ):
+    @bottleneck_switch()
+    def reduction(values, axis=None, skipna=True):
+        values, mask, dtype, dtype_max = _get_values(
+            values,
+            skipna,
+            fill_value_typ=fill_value_typ,
+        )
+
         if ((axis is not None and values.shape[axis] == 0)
                 or values.size == 0):
             try:
-                result = ensure_float(values.sum(axis, dtype=dtype_max))
+                result = getattr(values, meth)(axis, dtype=dtype_max)
                 result.fill(np.nan)
             except:
                 result = np.nan
         else:
-            result = values.min(axis)
+            result = getattr(values, meth)(axis)
 
-    result = _wrap_results(result, dtype)
-    return _maybe_null_out(result, axis, mask)
+        result = _wrap_results(result, dtype)
+        return _maybe_null_out(result, axis, mask)
```

Comment:
We have already wrapped the types by the time we call maybe_null_out. The result will already be coerced so I think the `is` check is safe.